crates/jql-parser/src/parser.rs

```diff
@@ -33,32 +32,29 @@ use crate::{
 };
 
 /// Parses the provided input and map it to the first matching token.
-fn parse_fragment(input: &str) -> IResult<&str, Token> {
+fn parse_fragment<'a>(input: &mut &'a str) -> PResult<Token<'a>> {
     alt((
```

Comment:
When all (or most performance critical) elements of a `alt` have unique first characters, `dispatch!` can offer some big performance wins

e.g. see https://github.com/toml-rs/toml/blob/main/crates/toml_edit/src/parser/value.rs#L18

Reply:
Sounds really good! I've just switched to it in a local branch and it works like a charm.